sklearn/ensemble/_iforest.py

```diff
@@ -327,6 +329,18 @@ def fit(self, X, y=None, sample_weight=None):
             check_input=False,
         )
 
+        self._average_path_length_max_samples = _average_path_length(
+            [self._max_samples]
+        )
+
+        self._average_path_length_per_tree = defaultdict()
+        self._decision_path_lenghts = defaultdict()
```

Comment:
Instead of dicts, I would rather use Python lists using the position of a tree in the `self.estimators_` list as positional index to access the precomputed path lengths.

This would avoid holding extra references to the tree objects in the keys of those dicts. I suspect that those extra references add complexity to the model serialization logic, especially for third party libraries for skops.